viewing_party/party.py

```diff
@@ -1,23 +1,189 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        new_movie = {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+        return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if title in movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            
+    return user_data
+            
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    total_ratings = 0
+    number_of_movies = 0
+
+    for movie_dict in user_data["watched"]:
+        total_ratings += movie_dict["rating"]
+        number_of_movies += 1
+
+    if number_of_movies == 0:
+        average_rating = 0.0
+    else:
+        average_rating = total_ratings/number_of_movies
 
+    return average_rating
+
+# my_data = {"watched": [{"title": "scary movie", "genre": "horror", "rating": "3"}, {"title": "another scary movie", "genre": "horror", "rating": "2"}, {"title": "funny movie", "genre": "comedy", "rating": "5"}]}
+def get_most_watched_genre(user_data):
+    genre_freq_dict = {}
+    
+    for movie_dict in user_data["watched"]:
+
+        if movie_dict["genre"] not in genre_freq_dict:
+            genre_freq_dict[movie_dict["genre"]] = 1
+        else:
+            genre_freq_dict[movie_dict["genre"]] += 1
+    
+    if user_data["watched"] == []:
+        max_genre = None
+    else:
+        max_genre = max(genre_freq_dict, key=genre_freq_dict.get)
+
+    return max_genre
+  
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+
+def get_unique_watched(user_data):

```

Comment:
Good job! Sets are a great way to remove duplicates and check the difference between two lists. Unfortunately, dictionaries can't be put in a set. In this case, it may be easier to use a for loop to see if a dictionary is in a list. For example, this is a potential solution:
```
for movie in user_movies:
        if movie not in friend_movies:
            user_unique_movie_titles.append(movie)
```
where `user_movies` is the user's movies and `friend_movies` is the list of all of the movies from all of the friends. This comment applies for the next function as well.